lib/rules/forbid-component-props.js

```diff
@@ -115,7 +142,9 @@ module.exports = {
       const prop = propName || propPattern;
       const options = {
         allowList: typeof value === 'string' ? [] : (value.allowedFor || []),
+        allowPatternList: typeof value === 'string' ? [] : value.allowedForPatterns || [],
         disallowList: typeof value === 'string' ? [] : (value.disallowedFor || []),
+        disallowPatternList: typeof value === 'string' ? [] : value.disallowedForPatterns || [],
```

Comment:
```suggestion
        allowPatternList: [].concat(value.allowedForPatterns || []),
        disallowList: [].concat(value.disallowedFor || []),
        disallowPatternList: [].concat(value.disallowedForPatterns || []),
```
altho i'm not sure why any of these would be able to be strings - doesn't the schema prevent that?

Reply:
Yeah, you're right about schema, but there is a [default](https://github.com/jsx-eslint/eslint-plugin-react/pull/3805/files/40beeb52feaf2989486565a3df98f2415e1aa5d3#diff-95ccf9a24cad95dd24cdc5d59460de65a598a88ebe8c19a2ee6af25978549eb8L16) also for this rule, so that's why this code supports strings also.

Reply:
I suppose the default could be made into an array, but nbd.